Playground for the new commondata format.ipynb

```diff
@@ -483,20 +505,198 @@
     "data_df"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": 61,
+   "id": "41394702",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dataclasses import dataclass\n",
+    "from pathlib import Path\n",
+    "import typing\n",
+    "\n",
+    "from validobj.custom import Parser\n",
+    "from validobj import ValidationError, parse_input\n",
+    "\n",
+    "\n",
+    "from validphys import convolution\n",
+    "\n",
+    "\n",
+    "# Scalar parsers\n",
+    "@Parser\n",
+    "def ValidPath(path_str: str) -> Path:\n",
+    "    \"\"\"Parse strings into paths\"\"\"\n",
+    "    try:\n",
+    "        return Path(path_str)\n",
+    "    except exception as e:\n",
+    "        raise ValidationError(f\"{path_str} is not a valid path\") from e\n",
+    "        \n",
+    "@Parser\n",
+    "def ValidOperation(op_str: str) -> str:\n",
+    "    \"\"\"Ensures that the operation defined in the commondata file is implemented in validphys\"\"\"\n",
+    "    ret = op_str.upper()\n",
+    "    if ret not in convolution.OP:\n",
+    "        raise ValidationError(f\"The operation '{op_str}' is not implemented in validphys\")\n",
+    "    return ret\n",
+    "\n",
+    "\n",
+    "\n",
+    "# Object classes wrappers\n",
+    "@dataclass\n",
+    "class TheoryMeta:\n",
+    "    FK_tables: list\n",
+    "    operation: ValidOperation\n",
+    "    conversion_factor: float = 1.0\n",
+    "    apfelcomb: dict = None\n",
```

Comment:
This should be Optional[dict]. And probably more structured.

Reply:
Plan is to remove all dictionaries (if that's what you mean with the structure)

Reply:
Yes, but also that as it is the default does not type check.